plugins/modules/aci_cloud_ctx_profile.py

```diff
@@ -0,0 +1,234 @@
+    - name of vrf to be managed
+    type: str
+  region:
+    description:
+    - name of region to be managed
+    type: str
+  vpn_gateway:
+    description:
+    - whether vpn gateway router is enabled or not
+    type: bool
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
+    type: str
+    default: present
+
+extends_documentation_fragment:
+- cisco.aci.aci
+'''
+
+EXAMPLES = r'''
+- name: Add a new aci cloud ctx profile
+  aci_cloud_ctx_profile:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    tenant: tenant_1
+    name: cloud_ctx_profile
+    vrf: VRF1
+    region: us-west-1
+    primary_cidr: '10.0.10.1/16'
+    state: present
+  delegate_to: localhost
+
+- name: Remove an aci cloud ctx profile
+  aci_cloud_ctx_profile:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    tenant: tenant_1
+    name: cloud_ctx_profile
+    state: absent
+  delegate_to: localhost
+
+- name: Query aci cloud ctx profile
+  aci_cloud_ctx_profile:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    tenant: anstest
+    name: ctx_profile_1
+    state: query
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        {
+            'descr': dict(type='str',),
+            'name': dict(type='str', aliases=['cloud_context_profile']),
+            'nameAlias': dict(type='str',),
+            'type': dict(type='str', choices=['regular', 'shadow'], ),
+            'tenant': dict(type='str',),
+            'state': dict(type='str', default='present', choices=['absent', 'present', 'query']),
+            'primary_cidr': dict(type='str',),
+            # FIXME: didn't find the flow_log in UI
+            # 'flow_log': dict(type='str'),
+            'vrf': dict(type='str'),
+            'region': dict(type='str'),
+            'vpn_gateway': dict(type='bool', default=False)
+        }
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['name', 'tenant']],
+            ['state', 'present', ['name', 'tenant', 'vrf', 'region', 'primary_cidr']],
+        ],
+    )
+
+    descr = module.params['descr']
+    name = module.params['name']
+    nameAlias = module.params['nameAlias']
+    type = module.params['type']
+    tenant = module.params['tenant']
+    state = module.params['state']
+    primary_cidr = module.params['primary_cidr']
+    child_configs = []
+
+    vrf = module.params['vrf']
+    region = module.params['region']
+    vpn_gateway = module.params['vpn_gateway']
+    if vrf:
```

Comment:
this 3 sections are only needed in present case. We should move it there. Also in present case, region and vrf are mandatory so not sure if we need the condition.